google/cloud/spanner_v1/table.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""User friendly container for Cloud Spanner Table."""
+
+_GET_SCHEMA_TEMPLATE = "SELECT * FROM {} LIMIT 0"
+
+
+class Table(object):
+    """Representation of a Cloud Spanner Table.
+    """
+
+    def __init__(self, table_id, database):
+        self.table_id = table_id
+        self._database = database
+
+    def get_schema(self):
```

Comment:
I think instead of having `get_schema`, it would make more sense to match the rest of the library and the BigQuery implementation by having this as a property. What do you think?

Reply:
That would be more consistent. My main worry with a property is it might be less clear that `table.schema` makes an API call compared to `table.get_schema()`.

Reply:
That is a good point. For the other resource objects (i.e. `Instance`, `Backup`, `Database`) we store local copies of properties and have a `reload()` method to make an RPC call to update them. I think it may make sense to stick to this design, especially if we decide to support create, update, and delete on the `Table` object.

I would initialize a private schema attribute as `None` and then call `reload` in `schema` if it is `None`. This means the property would make one API call at most. Then, if the user is expecting the schema to change frequently, they can all `reload` themselves to ensure the schema remains in sync with production. WDYT?

Reply:
> I would initialize a private schema attribute as None and then call reload in schema if it is None. This means the property would make one API call at most. Then, if the user is expecting the schema to change frequently, they can all reload themselves to ensure the schema remains in sync with production. WDYT?

I think that's a good approach. In BigQuery we didn't have this logic, so customers would surprisingly get `None` values, which caused all sorts of issues. I'll make this a property and add a reload method.

Reply:
Updated. Also added a `exists()` method to match [Database.reload()](https://googleapis.dev/python/spanner/latest/database-api.html#google.cloud.spanner_v1.database.Database.reload)'s `NotFound` exception behavior.